lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -77,11 +79,23 @@ public static Parser create() {
     return new Parser(state);
   }
 
-  public Tree parse(CharSequence input) {
+  public long isIdentOrOperator(CharSequence input) {
     byte[] inputBytes = input.toString().getBytes(StandardCharsets.UTF_8);
     ByteBuffer inputBuf = ByteBuffer.allocateDirect(inputBytes.length);
     inputBuf.put(inputBytes);
-    var serializedTree = parseInput(state, inputBuf);
+
+    return isIdentOrOperator(inputBuf);
+  }
+
+  public ByteBuffer parseInput(CharSequence input) {
```

Comment:
I knew we would have to make this method public. It makes the `Parser` API uglier - so far there was just a single `parse` method returning `Tree` now there are multiple ways to use this class... but let's be it for now.